src/main/java/macrobase/analysis/pipeline/BasicBatchedPipeline.java

```diff
@@ -57,15 +59,22 @@ public AnalysisResult contextualAnalyze(DataIngester ingester) throws SQLExcepti
     AnalysisResult run() throws SQLException, IOException, ConfigurationException {
         long startMs = System.currentTimeMillis();
         DataIngester ingester = conf.constructIngester();
+        // TODO: this should be a new pipeline
+        // Needs to happen early before ingester is possibly consumed. Downstream stages are allowed to drain
+        // upstream stages in construction.
+        if(contextualEnabled){
+            return contextualAnalyze(ingester);
+        }
+
         TimedBatchIngest batchIngest = new TimedBatchIngest(ingester);
         FeatureTransform featureTransform = new BatchScoreFeatureTransform(conf, batchIngest, conf.getTransformType());
         OutlierClassifier outlierClassifier = new BatchingPercentileClassifier(conf, featureTransform);
+        if (conf.getBoolean(MacroBaseConf.CLASSIFIER_DUMP, MacroBaseDefaults.CLASSIFIER_DUMP)) {
```

Comment:
not really. this class only outputs the result of the classification, which I don't care about.
I care about dumping metrics on datum before the transformation and after the transformation together. There is no place in the pipeline that we currently have access to both.
I guess now I can dump both transforms and classification results, and then combine, but this PR is not letting me do that, I still have to write custom code to accomplish that